From 1a87fce33c0ba9d230940a7787cd5cc0776de3a7 Mon Sep 17 00:00:00 2001
From: Matthias Kuhn <matthias@opengis.ch>
Date: Wed, 6 Feb 2019 10:26:53 +0100
Subject: [PATCH] Do not remove parameters from virtual methods which should be
 used in python

These methods did not actually override their parent methods but just coexisted and were never called.

Fixes compilation with SIP version: 4.19.14
---
 python/gui/auto_generated/qgsattributeformeditorwidget.sip.in   | 2 +-
 .../auto_generated/qgsattributeformrelationeditorwidget.sip.in  | 2 +-
 src/gui/qgsattributeformeditorwidget.h                          | 2 +-
 src/gui/qgsattributeformrelationeditorwidget.h                  | 2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/python/gui/auto_generated/qgsattributeformeditorwidget.sip.in b/python/gui/auto_generated/qgsattributeformeditorwidget.sip.in
index 19d154633e4..5e8d4524016 100644
--- a/python/gui/auto_generated/qgsattributeformeditorwidget.sip.in
+++ b/python/gui/auto_generated/qgsattributeformeditorwidget.sip.in
@@ -38,7 +38,7 @@ Constructor for QgsAttributeFormEditorWidget.
 
     ~QgsAttributeFormEditorWidget();
 
-    virtual void createSearchWidgetWrappers();
+    virtual void createSearchWidgetWrappers( const QgsAttributeEditorContext &context = QgsAttributeEditorContext() );
 
 
     void initialize( const QVariant &initialValue, bool mixedValues = false );
diff --git a/python/gui/auto_generated/qgsattributeformrelationeditorwidget.sip.in b/python/gui/auto_generated/qgsattributeformrelationeditorwidget.sip.in
index 8dd03ccebb0..73507143eaa 100644
--- a/python/gui/auto_generated/qgsattributeformrelationeditorwidget.sip.in
+++ b/python/gui/auto_generated/qgsattributeformrelationeditorwidget.sip.in
@@ -29,7 +29,7 @@ Widget to show for child relations on an attribute form.
 Constructor
 %End
 
-    virtual void createSearchWidgetWrappers();
+    virtual void createSearchWidgetWrappers( const QgsAttributeEditorContext &context = QgsAttributeEditorContext() );
 
     virtual QString currentFilterExpression() const;
 
diff --git a/src/gui/qgsattributeformeditorwidget.h b/src/gui/qgsattributeformeditorwidget.h
index 3d52ff2b1eb..6adab2915e5 100644
--- a/src/gui/qgsattributeformeditorwidget.h
+++ b/src/gui/qgsattributeformeditorwidget.h
@@ -57,7 +57,7 @@ class GUI_EXPORT QgsAttributeFormEditorWidget : public QgsAttributeFormWidget
 
     ~QgsAttributeFormEditorWidget() override;
 
-    void createSearchWidgetWrappers( const QgsAttributeEditorContext &context SIP_PYARGREMOVE = QgsAttributeEditorContext() ) override;
+    void createSearchWidgetWrappers( const QgsAttributeEditorContext &context = QgsAttributeEditorContext() ) override;
 
     /**
      * Resets the widget to an initial value.
diff --git a/src/gui/qgsattributeformrelationeditorwidget.h b/src/gui/qgsattributeformrelationeditorwidget.h
index e98ade78ab7..74b57ef558f 100644
--- a/src/gui/qgsattributeformrelationeditorwidget.h
+++ b/src/gui/qgsattributeformrelationeditorwidget.h
@@ -41,7 +41,7 @@ class GUI_EXPORT QgsAttributeFormRelationEditorWidget : public QgsAttributeFormW
      */
     explicit QgsAttributeFormRelationEditorWidget( QgsRelationWidgetWrapper *wrapper, QgsAttributeForm *form );
 
-    void createSearchWidgetWrappers( const QgsAttributeEditorContext &context SIP_PYARGREMOVE = QgsAttributeEditorContext() ) override;
+    void createSearchWidgetWrappers( const QgsAttributeEditorContext &context = QgsAttributeEditorContext() ) override;
     QString currentFilterExpression() const override;
 
   private:
